registry/consul/registry.go

```diff
@@ -0,0 +1,186 @@
+package consul
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"sync"
+	"time"
+
+	consul "github.com/hashicorp/consul/api"
+	"github.com/moderntv/cadre/registry"
+	"google.golang.org/grpc/grpclog"
+)
+
+var logger = grpclog.Component("consul_registry")
+
+var _ registry.Registry = &consulRegistry{}
+
+type consulRegistry struct {
+	client *consul.Client
+	// datacenter represents Consul datacenter.
+	datacenter string
+	// refreshPeriod determines period for Consul catalog API call.
+	refreshPeriod time.Duration
+	// aliases is used for mapping gRPC service names to Consul service names.
+	aliases map[string]string
+	mu      sync.RWMutex
```

Comment:
This should be either first or last

Reply:
If you are talking about the mutex I would object. Our conventions require that the mutex is defined above struct members it is protecting [conventions](https://moderntv.atlassian.net/wiki/spaces/BACK/pages/2121564180/Code+style+guide#Locking).